arelle/plugin/validate/NL/rules/fr_nl.py

```diff
@@ -163,6 +163,69 @@ def rule_fr_nl_1_01(
                         return
 
 
+@validation(
+    hook=ValidationHook.XBRL_FINALLY,
+    disclosureSystems=[
+        DISCLOSURE_SYSTEM_NT16,
+        DISCLOSURE_SYSTEM_NT17,
+        DISCLOSURE_SYSTEM_NT18
+    ],
+)
+def rule_fr_nl_2_04(
+        pluginData: PluginValidationDataExtension,
+        val: ValidateXbrl,
+        *args: Any,
+        **kwargs: Any,
+) -> Iterable[Validation] | None:
+    """
+    FR-NL-2.04: The 'link:schemaRef' element MUST NOT appear more than once
+    """
+    schema_ref_model_objects = []
+    for doc in val.modelXbrl.urlDocs.values():
+        if doc.type == ModelDocument.Type.INSTANCE:
+            for refDoc, docRef in doc.referencesDocument.items():
+                if docRef.referringModelObject.localName == "schemaRef":
+                    schema_ref_model_objects.append(docRef.referringModelObject)
```

Comment:
I'm personally a fan of yielding an error on each occurrence when possible. It's more detail and allows us to specify file and line number if available.

Reply:
The modelObjects field that we pass on validations can take a single object or a list of objects. For a list of objects it will add hrefs to the validation so that each object is represented with file and line numbers. In this case we can get locators to several things all at once instead of having to look at individual validations for those. In most cases, I agree with your sentiment that a single validation for a single modelObject is appropriate. But in this case, I think all offending schema refs should be highlighted in a single validation.